docs/articles/movie_recommendation_using_vectordb.markdown

```diff
@@ -0,0 +1,347 @@
+    model.train()
+    running_loss = 0.0
+    for inputs, labels in train_loader:
+        inputs, labels = inputs.to(device), labels.to(device)  # Move data to device
+        optimizer.zero_grad()
+        outputs = model(inputs)
+        loss = criterion(outputs, labels)
+        loss.backward()
+        optimizer.step()
+        running_loss += loss.item() * inputs.size(0)
+    epoch_loss = running_loss / len(train_loader.dataset)
+    print(f'Epoch [{epoch + 1}/{epochs}], Loss: {epoch_loss:.4f}')
+```
+
+That's it! We've successfully trained a Neural network for our genre classification task. You can check if the model is performing well or not in terms of the genre classification.
+
+```python
+from sklearn.metrics import f1_score
+
+model.eval()
+with torch.no_grad():
+    X_test_tensor, y_test_tensor = X_test_tensor.to(device), y_test_tensor.to(device)  # Move test data to device
+    outputs = model(X_test_tensor)
+    test_loss = criterion(outputs, y_test_tensor)
+    print(f'Test Loss: {test_loss.item():.4f}')
+
+
+thresholds = [0.1] * len(mlb.classes_)
+thresholds_tensor = torch.tensor(thresholds, device=device).unsqueeze(0)
+
+# Convert the outputs to binary predictions using varying thresholds
+predicted_labels = (outputs > thresholds_tensor).cpu().numpy()
+
+# Convert binary predictions and actual labels to multi-label format
+predicted_multilabels = mlb.inverse_transform(predicted_labels)
+actual_multilabels = mlb.inverse_transform(y_test_np)
+
+# Print the Predicted and Actual Labels for each movie
+for i, (predicted, actual) in enumerate(zip(predicted_multilabels, actual_multilabels)):
+    print(f'Movie {i+1}:')
+    print(f'    Predicted Labels: {predicted}')
+    print(f'    Actual Labels: {actual}')
+
+
+# Compute F1-score
+f1 = f1_score(y_test_np, predicted_labels, average='micro')
+print(f'F1-score: {f1:.4f}')
+
+# Saving the trained model
+torch.save(model.state_dict(), 'trained_model.pth')
+```
+
+## A movie recommendation system
+
+To create our movie recommendation system, we'll implement a process where users input a movie title, and we generate relevant recommendations based on that input. Key to this approach is utilizing Doc2Vec embeddings, which we'll store in a vector database for efficient retrieval.
+
+Here's how it works: When a user enters a movie title as a query, we'll first retrieve its embeddings from our vector database. With these embeddings in hand, our next step involves identifying 'n' number of movies whose embeddings closely resemble those of our query movie. We can achieve this by employing similarity metrics such as cosine similarity or by determining the smallest Euclidean distances. By leveraging these techniques, our system will provide users with personalized movie recommendations that are closely aligned with their initial movie interests. This approach ensures an intuitive and effective recommendation process tailored to individual preferences based on the genres.
+
```

Comment:
@vipul-maheshwari 
according to your discussion, all we have from the user is their query. Here for the first time you introduce "initial movie interests" and "individual preferences based on genres" - where do these come from? They're not in your code or discussion.

Reply:
Yes! So let's say If a user searches for 'Toy Story,' it indicates their interest in movies that fall under the genres of "Animation" or "Comedy"